jupyterhub_moss/spawner.py

```diff
@@ -220,6 +221,13 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
             )[0]
             options["environment_path"] = default_venv["path"]
 
+        for env in self.partitions[partition]["jupyter_environments"].values():
```

Comment:
I don't understand why you need to cycle though the `jupyter_environments` there. Also, I find the `for, else` syntax a bit obscure.

Reply:
It's checking whether the path corresponds to a default `jupyter_environments` or to a custom one.
As it is done now, only the `environment_path` is passed through the form, so it checks whether this path corresponds to one of the default envs, thus the loop.

If it's a custom one, then it is added to the `PATH`, if it's one of the defaults it checks `add_to_path`.

The `for else` syntax is bit uncommon but it is meant for this case. I'll propose an alternate implementation.